fix #535 observer in hosted games Tooltip

GrotheFAF opened the pull request:
collecting observers in a separate list not as one of the teams

- [ ] PR branch should be named `issuenum`-`fix`/`feature`/`cleanup`-`description`
- [ ] Code is split into logical commits
- [ ] Code has tests
- [ ] Final commit includes "Fixes #issue" in commit message

When all builds pass and a maintainer is happy with your PR, the "ready" label will be applied. Please complete these tasks then:

- [ ] Rebase onto develop
- [ ] Add changelog entry
- [ ] Remove this entire template section

GrotheFAF commented:
I don't know #709 (and I don't know what backported means)

Wesmania commented:
It's similar to changes I made in 537eccde90a3e5f57158dcc2161f86886f72ba07 in #709, I assumed you lifted it from there. Nevermind.

GrotheFAF commented:
How did you get to such an assuption? And why would I do that? You're pretty twisted.
Remember #591 ?

Wesmania commented on src/games/gameitem.py:
```diff
@@ -298,58 +297,50 @@ def update(self, message, old_client=None):
         teamlist     = []
 
         teams_string = ""
+        for i, team in enumerate(teams):
```
Adding to that previous comment - I checked out of interest, you actually can enumerate from something else than 0 if you give it the starting number as the second argument. Didn't even know that.

Wesmania commented:
It does seem that the team names sent from the server are strings instead of ints, and u'null'. That's so weird.

GrotheFAF committed:
fix #535 observer in hosted games Tooltip

observer are currently in team '-1' and 'null' (string)
collecting observers in a separate list not as one of the teams
redone after refactoring
add changelog

Wesmania commented:
+1, just edit the commit message to be more informative

GrotheFAF commented:
done 
(didn't realized that that would overwrite the old one message)

GrotheFAF commented:
add changelog

GrotheFAF committed:
Merge branch 'develop' into fix-observer-game-tooltip

GrotheFAF commented:
it is merged.
(a commit during merge, prevented to show it as merged)
